ceph/rados/pool_workflows.py

```diff
@@ -134,12 +150,14 @@ def fill_omap_entries(self, pool_name, **kwargs):
             deep_scrub_status = self.scrub_obj.verify_scrub_deepscrub(
                 before_deep_scrub_log, after_deep_scrub_log, "deepscrub"
             )
+
             if deep_scrub_status:
-                log.debug(
-                    "deep scrubbing has not been completed on the pool"
-                    "Sleeping for 10 seconds and checking again"
-                )
-                time.sleep(10)
+                log.info("deep scrubbing has not been completed on the pool")
+                if check_omap_entires():
```

Comment:
Is it intended that every time we call ` check_omap_entires `, the primary OSD should be restarted?

Reply:
Asking as if the Omap was not found, we sleep and continue. So the OSD could be restarted multiple times during the duration of sleep here.

Reply:
Modified to logic to first check if omap entries are available, and restart primary OSDs only in case of non-availability
Yes, agreed, Primary OSD will be restarted multiple times, this is because we do not know at which point deep-scrubbing actually reads and recognizes the OMAPs